### ISSUE TITLE
dropTargets need to be improved

### ISSUE THREAD

**Issue by ArnoldSmith86** [ArnoldSmith86/virtualtabletop#1205]
We played Animal Race last weekend and the recent changes to dropTargets were pretty annoying:

![image](https://user-images.githubusercontent.com/73538315/165765435-d5897604-1622-47ae-b114-e93624ea392b.png)

Of course everyone learned how to hit the hand after two failed attempts but when dragging a pile handle, you are forced to drag at the corner of the card which makes the condition "holder the mouse is over" not that great.

Please don't read this as me being annoyed about anything other than actually playing the game. I just want to start an issue so we can brainstorm ideas.

Personally, I think we should go back to only requiring the card to overlap the holder slightly and only use the new code (which was solving other issues) if the card overlaps multiple valid dropTargets. But @robartsd might have good examples why that doesn't make any sense.

**Comment by robartsd** [comment]
I agree that more improvement can be made. The intention is to make it so that only a little overlap is required, but also to require actual overlap, not just overlap of bounding client rectangles. We compromised both a bit. If the widget being moved is large and rotated or has rounded corners, it is easy for the overlapping bounding client rectangles method to lead to false positives.

**Comment by ArnoldSmith86** [comment]
Maybe we just add a special case for non-rotated holders?

**Comment by RaphaelAlvez** [comment]
> One of the things I've been struggling with is how to deal with rounded corners. I'm currently thinking that approximating them as polygons would be good enough, though I'm not sure on how many sides. 

No idea of the code you are talking about but parametric equations for circles and rectangles are so simple.

Why use something more complicated?

**Comment by robartsd** [comment]
> > One of the things I've been struggling with is how to deal with rounded corners. I'm currently thinking that approximating them as polygons would be good enough, though I'm not sure on how many sides.
> 
> No idea of the code you are talking about but parametric equations for circles and rectangles are so simple.
> 
> Why use something more complicated?

For testing a single point, parametric equations would work well enough. The difficulty comes in optimizing to test if ANY point in one shape is in another shape.
